lib/spack/spack/package.py

```diff
@@ -202,14 +202,19 @@ def determine_spec_details(cls, prefix, exes_in_prefix):
                         external_modules = extra_attributes.pop(
                             'modules', None
                         )
-                        spec = spack.spec.Spec(
-                            spec_str,
-                            external_path=external_path,
-                            external_modules=external_modules
-                        )
-                        specs.append(spack.spec.Spec.from_detection(
-                            spec, extra_attributes=extra_attributes
-                        ))
+                        try:
+                            spec = spack.spec.Spec(
+                                spec_str,
+                                external_path=external_path,
+                                external_modules=external_modules
+                            )
+                        except Exception as e:
```

Comment:
Is there a more specific exception we can catch?

Reply:
No, I think it should swallow everything and report. We want to avoid errors like the one in the issue, where probably the user was not even interested in `fzf`